File: PS/Programmers/구명보트.py
def solution(people, limit):
    answer = 0
    sorted(people)
    right = len(people)-1
    left = 0
    while(left < right+1):
        if (people[right]+people[left]) <= limit:
            left+=1
        right-=1
        answer+=1
        print(str(left) +" " + str(right))
    return answer

File: PS/Programmers/구명보트.cc
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
int solution(vector<int> people, int limit) {
	int answer = 0;
	sort(people.begin(), people.end());
	int right = people.size()-1, left =0;
	while (left < right+1)
	{
		if (people[right] + people[left] <= limit)
			left++;
		right--;
		answer++;
	}
	return answer;
}